### PR TITLE
CLID-8: Initial E2E test for v2

### PR DESCRIPTION
**Pull request by sherine-k**
# Description
This PR introduces end to end tests for oc-mirror v2 for scenarios:
* additional image mirroring from mirror to disk + from disk to mirror
* release image mirroring from mirror to disk + from disk to mirror

All unit tests are run thanks to a `-short` flag in the `go test` command in Makefile
All E2E tests have to be run sequentially, using a `-run $NAME_OF_TEST_FUNCTION` in the Makefile. 

A few changes and fixes are included in this PR as well, essentially to fix some race conditions encountered while running the E2E tests:
* Add a [mutex](https://github.com/openshift/oc-mirror/blob/d4a922e0eb5e61924b49f42f8abf5c1633b7d6a7/v2/pkg/batch/worker.go#L90-L92) around the errArray used by batch workers
* Provide a way to [override the system registries.conf](https://github.com/openshift/oc-mirror/blob/d4a922e0eb5e61924b49f42f8abf5c1633b7d6a7/v2/pkg/cli/executor.go#L274-L276) location 
* Use the [cobra's out writer](https://github.com/openshift/oc-mirror/blob/d4a922e0eb5e61924b49f42f8abf5c1633b7d6a7/v2/pkg/cli/executor.go#L201-L205) when in debug mode for `containers/image's copy` , in order to avoid creating log files concurrently on disk
* Moved the setup of `opts.RemoveSignatures, _ = strconv.ParseBool("true")` outside of the `mirror.Run` so that it doesn't get called within the batch workers goroutines.

Fixes [CLID-8](https://issues.redhat.com//browse/CLID-8)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Prow CI passes

### CODE REVIEW

File: v2/internal/testutils/testutils.go [hunk 1]
```diff
@@ -122,3 +131,276 @@ func LocalMirrorFromFiles(source string, destination string) error {
...
+	}
+	c := map[string][]byte{
+		"/release-manifests/image-references": imageRefsBytes,
+		"/release-manifests/release-metadata": releaseMetaBytes,
+	}
+	tag, err := name.NewTag(releaseImgRef)
+	if err != nil {
+		return "", err
+	}
+	i, _ := crane.Image(c)
```
Comment (lmzuccarelli) [inline v2/internal/testutils/testutils.go:324]: @sherine-k - looks like some duplication here - maybe create a simple common utility

**Comment by lmzuccarelli** [comment]
@sherine-k - awesome stuff,  allows us to debug our e2e tests (as @aguidirh mentioned), improves confidence in our services, which at the end of the day is what we are delivering to our clients

My next steps is to verify locally and do some delve debugging :)

**Comment by lmzuccarelli** [comment]
@sherine-k  I'll also wait for @aguidirh review.

Just left a comment about duplicate code blocks

File: v2/internal/testutils/testutils.go (hunk 1 above)
Comment (sherine-k) [reply v2/internal/testutils/testutils.go]: That's very good point thanks!

**Pull request merged**